near-sdk-macros/src/lib.rs

```diff
@@ -207,6 +209,174 @@ pub fn metadata(item: TokenStream) -> TokenStream {
     }
 }
 
+#[cfg(feature = "abi")]
+#[proc_macro_derive(NearSchema, attributes(abi, serde, borsh_skip, schemars, validate))]
+pub fn derive_near_schema(input: TokenStream) -> TokenStream {
+    let mut input = syn::parse_macro_input!(input as syn::DeriveInput);
+
+    let (mut json_schema, mut borsh_schema) = (false, false);
+    let mut type_attrs = vec![];
+    for attr in input.attrs {
+        match attr.parse_meta() {
+            // #[abi]
+            Ok(syn::Meta::Path(meta)) if meta.is_ident("abi") => {
+                return TokenStream::from(
+                    syn::Error::new_spanned(
+                        meta.into_token_stream(),
+                        "attribute requires at least one argument",
+                    )
+                    .to_compile_error(),
+                );
+            }
+            // #[abi(json, borsh)]
+            Ok(syn::Meta::List(meta)) if meta.path.is_ident("abi") => {
+                // #[abi()]
+                if meta.nested.is_empty() {
+                    return TokenStream::from(
+                        syn::Error::new_spanned(
+                            meta.into_token_stream(),
+                            "attribute requires at least one argument",
+                        )
+                        .to_compile_error(),
+                    );
+                }
+                json_schema = false;
```

Comment:
why is this needed? Seems like a strange condition. Can you add a comment on why if it is?

Reply:
Will document, but for here, it's because omitting the `#[abi]` spec defaults to deriving only `JsonSchema`. `#[abi(borsh)]` derives only `BorshSchema`. And `#[abi(json, borsh)]` derives both.